Debug ShadowMap tests

ptrgags opened the pull request:
This PR re-enables a few tests that were `xit-ed` out in `ShadowMap` and fixes them so they run again. The problems were:

1. Due to rendering differences The rendering color wasn't an exact match in some cases
2. For the point light test, it wasn't waiting for the ready promise, so the boxes weren't showing up. I changed it to use a single box and just reposition it in the scene as needed.

@j9liu this is ready for review, though wait for #10695 to be merged before merging

cesium-concierge commented:
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -825,7 +865,9 @@ describe(
     });
 
     // TODO: Need to debug this test
```
Delete this comment

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -797,7 +827,17 @@ describe(
+
+      // Change the time so that the shadows are no longer pointing straight down
+      renderAndCall(function (rgba) {
+        // Due to shading differences, the result might not exactly the same, but
```
What does "shading differences" refer to here -- differences between new `Model` and old `Model`?

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -797,7 +827,17 @@ describe(
+        // it should be close
+        const [r, g, b, a] = colorAbsoluteDifference(rgba, unshadowedColor);
+        expect(r).toBeLessThan(48);
+        expect(g).toBeLessThan(4);
```
The epsilons seem arbitrary to me since they differ so much for each test... is there no way to make them more consistent? Why do we consider a color difference of 48 or even 64 "close" ?

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -797,7 +827,17 @@ describe(
       });
-      renderAndExpect(unshadowedColor);
+
+      // Change the time so that the shadows are no longer pointing straight down
```
Delete this comment since the time isn't being changed

j9liu reviewed (commented):
@ptrgags I left a review -- I feel iffy about the approach because the difference in the expected colors seems wildly different. Maybe we can change something about the test setup so the expectations aren't as arbitrary.

ptrgags commented:
@j9liu I updated those tests to be a bit more robust: 

- I changed the camera setup so it's pointing straight down above the floor to ensure it sees the shadow. 
- I noticed that one test didn't pass `startTime` to the render call, which explains why I was getting different results depending on when I was running it yesterday.
- I changed how the colors are checked to rely less on arbitrary thresholds.

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -818,25 +818,37 @@ describe(
       });
 
-      // Change the time so that the shadows are no longer pointing straight down
+      // Change the light direction so that the shadows are no longer pointing straight down
```
this is now a duplicate of the comment above :')

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -788,16 +818,38 @@ describe(
+        const red = rgba[0];
+        const unshadowedRed = unshadowedColor[0];
+        const shadowedRed = shadowedColor[0];
+        expect(shadowedRed).toBeLessThan(red);
```
same suggestion as above

```suggestion
        expect(red).toBeGreaterThan(shadowedRed);
```

j9liu commented on Specs/Scene/ShadowMapSpec.js:
```diff
@@ -732,23 +743,42 @@ describe(
+        const red = rgba[0];
+        const unshadowedRed = unshadowedColor[0];
+        const shadowedRed = shadowedColor[0];
+        expect(shadowedRed).toBeLessThan(red);
```
this may seem nitpicky, but I think this reads better:

```suggestion
        expect(red).toBeGreaterThan(shadowedRed);
```

j9liu reviewed (commented):
@ptrgags This looks a lot better :) just a couple tweaks

ptrgags commented:
@j9liu updated!

j9liu commented:
@ptrgags looks good, thanks!
